Wrapping/Generators/Python/CMakeLists.txt

```diff
@@ -1002,3 +1002,10 @@ else()
   endmacro()
 
 endif()
+
+add_custom_command(OUTPUT ${ITK_WRAP_PYTHON_BINARY_DIR}/itk/__init__.pyi
+    COMMAND ${Python3_EXECUTABLE} ${ITK_WRAP_PYTHON_BINARY_DIR}/itk_generate_pyi.py
+    DEPENDS ${ITK_WRAP_PYTHON_FILES} ${WRAPPER_LIBRARY_NAME}Swig
```

Comment:
I am concerned that these dependencies are insufficient for parallel builds.  This script needs to be the last thing that is run (AFTER all the libraries are created and all the python files are generated).  A fully working python environment is needed for the introspection to work.

Reply:
Yes, we should keep the parallel build capabilities. We could generate a `*.pyi` file for each module's wrapping. This will also make it possible to generate `*.pyi` files for remote modules. The script may need to be tweaked, but this `add_custom_command` could be added in the `itk_end_wrap_module_python` macro where the `${WRAPPER_LIBRARY_NAME}Swig` is already defined (note that there is one per module).  The module to generate types for would be `itk.ITKSmoothingPython`, `itk.ITKCommonPython`, `itk.ITKSpatialObjectsPython`, etc.. Then, I assume that there is a way to reference these from `itk/__init__.pyi`.